humps.js

```diff
@@ -40,26 +40,29 @@
   // String conversion methods
 
   var separateWords = function(string, separator) {
-    if (separator === undefined) {
+    if (typeof separator === 'undefined') {
       separator = '_';
     }
     return string.replace(/([a-z])([A-Z0-9])/g, '$1'+ separator +'$2');
   };
 
   var camelize = function(string) {
+    // efficiently checks if a key is a number
+    var number = string - 0;
+    if (number === number) {
```

Comment:
Could we use something like `Number(string)` to check this?

I understand `if (number === number)`, but it feels a little too strange